core/shared/src/main/scala/fs2/async/mutable/Queue.scala

```diff
@@ -40,12 +40,22 @@ trait Queue[F[_], A] { self =>
   /** Dequeue one `A` from this queue. Completes once one is ready. */
   def dequeue1: F[A]
 
+  /** Dequeue all available `A`s from this queue. Completes once any values are ready. */
+  def dequeueAvailable1: F[Vector[A]]
+
   /** Like `dequeue1` but provides a way to cancel the dequeue. */
   def cancellableDequeue1: F[(F[A], F[Unit])]
 
+  /** Like `dequeueAvailable1` but provides a way to cancel the dequeue. */
+  def cancellableDequeueAvailable1: F[(F[Vector[A]], F[Unit])]
+
   /** Repeatedly call `dequeue1` forever. */
   def dequeue: Stream[F, A] = Stream.bracket(cancellableDequeue1)(d => Stream.eval(d._1), d => d._2).repeat
 
+  /** Repeatedly call `dequeueAvailable1` forever. */
+  def dequeueAvailable: Stream[F, A] =
```

Comment:
In that case, a pushback addition to the queue would work - but is that better than the current code?